lib/pbench/server/utils.py

```diff
@@ -73,6 +75,11 @@ def quarantine(dest, logger, *files):
         if not os.path.exists(afile) and not os.path.islink(afile):
             continue
         try:
+            if str(afile).endswith(".tar.xz"):
+                try:
+                    Dataset.attach(path=afile, state=States.QUARANTINED)
+                except DatasetNotFound:
+                    logger.exception("quarantine dataset {} not found", afile)
             shutil.move(afile, os.path.join(dest, os.path.basename(afile)))
```

Comment:
Should we be deferring the state update until after the move is complete?

(This is as far as I got in the review.  Let me know if I should continue.)

Reply:
I considered which was "best" and decided I'd rather try to set the state even if the actual move failed, because we won't be able to operate on the dataset in the future. I could have put it in a `finally`, I suppose.

Reply:
Also, while reviewing the `gold/xxx.txt` files won't be very productive unless you're familiar with the flow of the unit tests, GitHub's sorting means there are plenty of product files hidden among the unit test changes. You can also gain some understanding (and join us in our regular migraine parties) by reading the `unittest` script changes to start to get some corruption (er, um, I mean "understanding") of the command line unit test framework.